pkg/ebpf/common.go

```diff
@@ -78,27 +79,52 @@ func verifyOSVersion(kernelCode uint32, platform string, exclusionList []string)
 		return true, nil
 	}
 
-	if isCentOS(platform) && kernelCode >= centOSMinRequiredKernelCode {
-		return true, nil
-	}
-
 	if isUbuntu(platform) {
 		if kernelCode >= linuxKernelVersionCode(4, 4, 119) && kernelCode <= linuxKernelVersionCode(4, 4, 126) {
 			return false, fmt.Errorf("got ubuntu kernel %s with known bug on platform: %s, see: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1763454", kernelCodeToString(kernelCode), platform)
 		}
 	}
 
-	if kernelCode < minRequiredKernelCode {
-		return false, fmt.Errorf(
-			"incompatible linux version. at least %s (%d) required, got %s (%d)",
-			kernelCodeToString(minRequiredKernelCode),
-			minRequiredKernelCode,
-			kernelCodeToString(kernelCode),
-			kernelCode,
-		)
+	kallsyms, err := readKallsyms()
+	if err != nil {
+		log.Warnf("error reading /proc/kallsyms file: %s", err)
+		// If we can't read the /proc/kallsyms file let's just return true to avoid blocking the tracer from running
+		return true, nil
+	}
+
+	return verifyKernelFuncs(kallsyms), nil
+}
+
+func readKallsyms() (string, error) {
+	procRoot := util.GetProcRoot()
+	raw, err := ioutil.ReadFile(path.Join(procRoot, "kallsyms"))
+	if err != nil {
+		return "", errors.Wrapf(err, "error reading kallsyms file from proc dir: %s", procRoot)
 	}
 
-	return true, nil
+	return string(raw), nil
+}
+
+func verifyKernelFuncs(kallsyms string) bool {
+	lines := strings.Split(kallsyms, "\n")
+	funcs := make(map[string]struct{}, len(lines))
+
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if len(fields) < 3 {
+			continue
+		}
+
+		funcs[fields[2]] = struct{}{}
+	}
+
+	for _, f := range requiredKernelFuncs {
```

Comment:
There might be a problem with this logic, if all lines have less than 3 fields, then your `funcs` is empty, then this `for` loop would not execute, and you returned `true` in the end, but in reality none of the lines contain `requiredKernelFuncs`, right?

Reply:
👍  I will change that

Reply:
actually this is already working (I added a test case for it though)